Extensions/ExtendedMath.json

```diff
@@ -0,0 +1,553 @@
+  "name": "ExtendedMath",
+  "shortDescription": "A general extention that adds additional math functions and constants expressions.",
+  "tags": "math, constants, conversions, colours, coordinates",
+  "version": "1.0.0",
+  "eventsFunctions": [
+    {
+      "description": "Returns a term from the Fibonacci sequence ",
+      "fullName": "Fibonacci numbers",
+      "functionType": "Expression",
+      "name": "fibonacci",
+      "sentence": "",
+      "events": [
+        {
+          "disabled": false,
+          "folded": false,
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "function fibonacci(n) {\n    let initial = 0;\n    let final = 1;\n    let sum = 0;\n\n    let i = 0;\n    for(i = 2; i <= n; i++) {\n        sum = initial + final;\n\n        initial = final;\n        final = sum;\n    }\n    return n ? final : initial;\n}\n\nconst n = eventsFunctionContext.getArgument(\"n\");\n\neventsFunctionContext.returnValue = fibonacci(n);",
+          "parameterObjects": "",
+          "useStrict": true
+        }
+      ],
+      "parameters": [
+        {
+          "codeOnly": false,
+          "defaultValue": "",
+          "description": "The desired term in the sequence",
+          "longDescription": "",
+          "name": "n",
+          "optional": false,
+          "supplementaryInformation": "",
+          "type": "expression"
+        }
+      ],
+      "objectGroups": []
+    },
+    {
+      "description": "Returns the specified mathematical constant (0 by default)",
+      "fullName": "Constant",
+      "functionType": "Expression",
+      "name": "constant",
+      "sentence": "",
+      "events": [
+        {
+          "disabled": false,
+          "folded": false,
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "function constants(n) {\r\n    n = n.toLowerCase();\r\n    n = n.replace(' ', '_');\r\n\r\n    switch (n) {\r\n        case 'φ':\r\n        case 'golden_ratio':\r\n            return 1.618033988749894;\r\n        case 'quarter_pi':\r\n            return Math.PI / 4;\r\n        case 'half_pi':\r\n            return Math.PI / 2;\r\n        case 'π':\r\n        case 'pi':\r\n            return Math.PI;\r\n        case 'τ':\r\n        case 'tau':\r\n        case 'two_pi':\r\n            return Math.PI * 2;\r\n        case 'sqrt1/2':\r\n        case 'sqrt1_2':\r\n            return Math.SQRT1_2;\r\n        case 'sqrt2':\r\n            return Math.SQRT2;\r\n        case 'log10e':\r\n            return Math.LOG10E;\r\n        case 'log2e':\r\n            return Math.LOG2E;\r\n        case 'ln10':\r\n            return Math.LN10;\r\n        case 'ln2':\r\n            return Math.LN2;\r\n        case 'e':\r\n            return Math.E;\r\n        default:\r\n            return 0;\r\n    }\r\n}\r\n\r\nconst constantName = eventsFunctionContext.getArgument(\"constantName\");\r\neventsFunctionContext.returnValue = constants(constantName); \r\n",
```

Comment:
Not a big fan of having all constants in a single function, it decreases the discoverability for the user, how do you even know the constant name? 

Instead, we should have a bunch of expression for each constant - JS engine will optimize this much easier, and it's easier to search in the list of expressions.

Reply:
(I know it's a pain, but this mimics what most programming language standard libraries do: Math.PI, Math.SQRT2...). 
In a game performance is important, so it's really a bad thing that everytime that I ask for PI, the function is:
- declaring a function (dozens of CPU cycles wasted)
- changing the string to lower case (dozens of CPU cycles again)
- replacing spaces (dozens of CPU cycles wasted again)
- making a switch (dozens of CPU cycles wasted again).

All of this for a single constant that should be a few CPU cycles. So don't complicate stuff, make an expression for each constant.
The function should be this: 

`eventsFunctionContext.returnValue = Math.PI`

Simple, efficient! :)

Reply:
The best thing being: if the user enter a wrong name, like "ExtendedMath.PA()", s/he will get an error. Instead of silently getting 0 and searching for half an hour why his/her game is not working ;)